sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_condition.py

```diff
@@ -36,13 +38,24 @@ def pipeline_func(int_param1: int, int_param2: int):
         )
 
     :param condition: The condition of the execution flow.
-        The value could be a boolean type control output or a pipeline expression.
-    :type condition: Union[str, bool, InputOutputBase]
+        The value could be a boolean type control output, a node with exactly one boolean type output
+        or a pipeline expression.
+    :type condition: Union[str, bool, InputOutputBase, BaseNode, PipelineExpression]
     :param true_block: Block to be executed if condition resolved result is true.
     :type true_block: BaseNode
     :param false_block: Block to be executed if condition resolved result is false.
     :type false_block:  BaseNode
     :return: The condition node.
     :rtype: ConditionNode
     """
+    # resolve expression as command component
+    if isinstance(condition, PipelineExpression):
+        condition = condition.resolve()
+    if isinstance(condition, BaseNode):
+        if len(condition.outputs) != 1:
```

Comment:
Should we validate if the output is control output?

Reply:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py#L61
seems we already have validation on this?